File: levelledmobs-plugin-bukkit/src/main/java/me/lokka30/levelledmobs/bukkit/listeners/PlayerJoinListener.java
package me.lokka30.levelledmobs.bukkit.listeners;

import me.lokka30.levelledmobs.bukkit.LevelledMobs;
import me.lokka30.levelledmobs.bukkit.logic.RunContext;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.player.PlayerJoinEvent;

public final class PlayerJoinListener extends ListenerWrapper {

    public PlayerJoinListener() {
        super(true);
    }

    @EventHandler(priority = EventPriority.MONITOR)
    public void handle(final PlayerJoinEvent event) {
        /* Trigger */
        LevelledMobs.getInstance().getLogicHandler().runFunctionsWithTriggers(
            new RunContext().withPlayer(event.getPlayer()),
            "on-player-join"
        );
    }
}

File: levelledmobs-plugin-bukkit/src/main/kotlin/me/lokka30/levelledmobs/bukkit/listeners/PlayerJoinListener.kt
package me.lokka30.levelledmobs.bukkit.listeners

import org.bukkit.ChatColor.AQUA
import org.bukkit.ChatColor.DARK_GRAY
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.player.PlayerJoinEvent

/*
FIXME comment
 */
class PlayerJoinListener : ListenerWrapper(
    "org.bukkit.event.player.PlayerJoinEvent"
) {

    @EventHandler(priority = EventPriority.MONITOR)
    fun handle(event: PlayerJoinEvent) {
        event.player.sendMessage("${AQUA}This server is running LevelledMobs 4 [Alpha]")
        event.player.sendMessage("${DARK_GRAY}https://github.com/lokka30/LevelledMobs/")
        //FIXME add more stuff
    }

}